Program.cs

```diff
@@ -135,20 +136,20 @@ void GameLoop()
                     builder.Clear();
                 }
 
-                m_playerLine[m_playerPosition] = m_playerCharacter;
+                m_playerLine[m_playerPosition] = m_playerChar;
                 Console.WriteLine("|-------|");
                 Console.WriteLine("|" + string.Concat(m_playerLine)+ "|");
 
                 currentGameTick++;
 
                 // Game End State
-                if (ShouldUpdateGameWorld(currentGameTick) 
+                if (currentGameTick % 5 == 0 
                         && m_obstacles.Count > 0
                         && m_obstacles.First().m_yPosition == m_height - 1
                         && m_playerPosition == m_obstacles.First().m_xPosition) 
                 {
                     Console.WriteLine("Collision Occurred!");
-                    m_GameEnded = true;
+                    break;

```

Comment:
I think this is a cleaner approach. Just breaking the loop when it's time to break. Rather than continuing the whole game loop before the end occurs. 

That said. I suppose there's value in saying that a game "loop" is always complete. I.e. you can always rely that the game loop will be completed in full every time. I can imagine a game loop that's super big and spooky and if there were many exit points then it could get complex. That could be a good thing though.